lib/Parser/Parser.cpp

```diff
@@ -203,6 +205,30 @@ Token Lexer::getNextToken(std::string &ErrStr) {
                         StringRef(NumBegin, NumEnd - NumBegin), 10)};
   }
 
+  if (*Begin == '(') {
+    ++Begin;
+    const char *NumBegin = Begin;
+    while (Begin != End && (*Begin == '0' || *Begin == '1' || *Begin == 'x')) {
+      ++Begin;
+    }
+    const char *NumEnd = Begin - 1;
+    if (Begin == NumBegin) {
+      ErrStr = "invalid knownbits string";
+      return Token{Token::Error, Begin, 0, APInt()};
+    }
+    Token T;
+    T.K = Token::KnownBits;
+    T.Pos = NumBegin;
+    T.Len = size_t(Begin - NumBegin);
+    T.PatternString = StringRef(NumBegin, NumEnd - NumBegin + 1); 
+    if (Begin != End && *Begin != ')') {
```

Comment:
probably this entire test goes a little earlier anyway-- make it part of the previous validity test, here it is happening confusingly late after you have built a token